### PR TITLE
Alot of fixes for FlxGameJolt.hx

### PR DESCRIPTION
**Pull request by jigsaw-4277821**
* Minor fixes
* Make returnMap to be available (still private tho), not only for a function
* Sys.args is not only for desktop so it works on all sys platforms

### CODE REVIEW

**Comment by Geokureli** [comment]
why was `verbose` removed?

File: flixel/addons/api/FlxGameJolt.hx [hunk 1]
```diff
@@ -182,17 +180,22 @@ class FlxGameJolt
...
+	 * @param 	UserName	The username to authenticate, if AutoAuth is true. If you set AutoAuth to true but don't put a value here, FlxGameJolt will attempt to get the user data automatically, which will only work for Flash embedded on GameJolt, or games run via Quick Play.
+	 * @param 	UserToken	The user token to authenticate, if AutoAuth is true. If you set AutoAuth to true but don't put a value here, FlxGameJolt will attempt to get the user data automatically, which will only work for Flash embedded on GameJolt, or games run via Quick Play.
 	 * @param 	Callback 	An optional callback function, which is only used if AutoAuth is set to true. Will return true if authentication was successful, false otherwise.
 	 */
 	public static function init(GameID:Int, PrivateKey:String, AutoAuth:Bool = false, ?UserName:String, ?UserToken:String, ?Callback:Dynamic):Void
 	{
 		if (_gameID != 0 && _privateKey != "")
+		{
+			Callback(false);
```
Comment (Geokureli) [inline flixel/addons/api/FlxGameJolt.hx:191]: will get a null ref if a callback is not passed in. 

also docs say:
> only used if AutoAuth is set to true

File: flixel/addons/api/FlxGameJolt.hx [hunk 2]
```diff
@@ -182,17 +180,22 @@ class FlxGameJolt
...
 			return;
+		}
 
 		_gameID = GameID;
 		_privateKey = PrivateKey;
+		
+		Callback(true);
```
Comment (Geokureli) [inline flixel/addons/api/FlxGameJolt.hx:198]: should also be null checked, but why are we calling it here? its going to get called again below if AutoAuth is true

File: flixel/addons/api/FlxGameJolt.hx [hunk 3]
```diff
@@ -691,8 +694,7 @@ class FlxGameJolt
 			_loader = new URLLoader();
 
 		#if debug
-		if (verbose)
-			FlxG.log.add("FlxGameJolt: Contacting " + request.url);
+		FlxG.log.add("FlxGameJolt: Contacting " + request.url);
```
Comment (Geokureli) [inline flixel/addons/api/FlxGameJolt.hx:697]: keep verbose check

File: flixel/addons/api/FlxGameJolt.hx [hunk 4]
```diff
@@ -738,7 +739,7 @@ class FlxGameJolt
 		}
 
 		#if debug
-		if (returnMap.exists("message") && verbose)
+		if (returnMap.exists("message"))
```
Comment (Geokureli) [inline flixel/addons/api/FlxGameJolt.hx:742]: keep verbose check

**Comment by Geokureli** [comment]
you'll also need to make a change to the Flixel-demo [FlxPongApi](https://haxeflixel.com/demos/FlxPongApi/).
it uses fetchScore with a `tableId` here: https://github.com/HaxeFlixel/flixel-demos/blob/bf84df3964e506dbb5e63b45884aeed88376f523/Arcade/FlxPongApi/source/MenuState.hx#L430

File: flixel/addons/api/FlxGameJolt.hx (hunk 2 above)
Comment (jigsaw-4277821) [reply flixel/addons/api/FlxGameJolt.hx:198]: Ok, I'll fix them

**Comment by jigsaw-4277821** [comment]
@Geokureli is good now?

File: flixel/addons/api/FlxGameJolt.hx [hunk 5]
```diff
@@ -427,12 +432,12 @@ class FlxGameJolt
 	 * @param	Limit		The maximum number of scores to retrieve. Leave null to retrieve only this user's scores.
```
Comment (Geokureli) [inline flixel/addons/api/FlxGameJolt.hx:432]: need to add tableID param to the documentation

**Review by Geokureli — CHANGES_REQUESTED** [review]
One more thing I forgot to mention

**Comment by jigsaw-4277821** [comment]
@Geokureli it's ok now?

**Review by Geokureli — APPROVED** [review]

File: flixel/addons/api/FlxGameJolt.hx [hunk 6]
```diff
@@ -423,17 +428,21 @@ class FlxGameJolt
...
 	 * Retrieve the high scores from this game's remote data. If not authenticated, leaving Limit null will still return the top ten scores. Requires initialization.
 	 *
-	 * @see		http://gamejolt.com/api/doc/game/scores/fetch/
+	 * @see		https://gamejolt.com/api/doc/game/scores/fetch/
+	 * @param	TableID		The table you want to pull from
 	 * @param	Limit		The maximum number of scores to retrieve. Leave null to retrieve only this user's scores.
 	 * @param	CallBack	An optional callback function. Will return a Map<String:String> whose keys and values are equivalent to the key-value pairs returned by GameJolt.
 	 */
-	public static function fetchScore(?Limit:Int, ?Callback:Dynamic):Void
+	public static function fetchScore(?TableID:Int, ?Limit:Int, ?Callback:Dynamic):Void
```
Comment (Geokureli) [inline flixel/addons/api/FlxGameJolt.hx:436]: is TableID optional? what happens if you don't specify it?

If it is optional it should be added after `Limit` for better backwards compatibility

**Review by Geokureli — CHANGES_REQUESTED** [review]

File: flixel/addons/api/FlxGameJolt.hx (hunk 6 above)
Comment (Geokureli) [reply flixel/addons/api/FlxGameJolt.hx:436]: Also, if optional the Doc should explain what happens if ommitted, as well

**Comment by jigsaw-4277821** [comment]
My explanation of it isn't the best, wanna help @Geokureli ?

File: flixel/addons/api/FlxGameJolt.hx [hunk 7]
```diff
@@ -429,7 +429,7 @@ class FlxGameJolt
 	 * Retrieve the high scores from this game's remote data. If not authenticated, leaving Limit null will still return the top ten scores. Requires initialization.
 	 *
 	 * @see		https://gamejolt.com/api/doc/game/scores/fetch/
-	 * @param	TableID		The table you want to pull from
+	 * @param 	TableID		Optional: the ID of the table you want to pull data from. If null, the pulled score will be returned as the highes. Ignored if zero.
```
Comment (Geokureli) [inline flixel/addons/api/FlxGameJolt.hx:432]: The ID of the table you want to pull data from. If left blank, the scores from the primary score table will be returned.

**Review by Geokureli — CHANGES_REQUESTED** [review]
TableID should be the second arg, for backwards compatibility
```hx
public static function fetchScore(?Limit:Int, ?TableID:Int, ?Callback:Dynamic):Void
```

File: flixel/addons/api/FlxGameJolt.hx (hunk 7 above)
Comment (jigsaw-4277821) [reply flixel/addons/api/FlxGameJolt.hx]: I just saw this thanks

**Comment by jigsaw-4277821** [comment]
> TableID should be the second arg, for backwards compatibility. Like so:
> 
> ```hx
> public static function fetchScore(?Limit:Int, ?TableID:Int, ?Callback:Dynamic):Void
> ```

Thank you

**Comment by jigsaw-4277821** [comment]
All are good now, you can push it or whatever

**Review by Geokureli — APPROVED** [review]

**Comment by Geokureli** [comment]
LOL

just noticed one more thing! gonna change it myself

**Comment by jigsaw-4277821** [comment]
Ok

**Comment by Geokureli** [comment]
I gotta run errors, I'll merge this when I get back if Github Actions are successful

**Comment by jigsaw-4277821** [comment]
Ok

**Comment by jigsaw-4277821** [comment]
@Geokureli flash fails

**Pull request merged**